### PR TITLE
Models

### PR DESCRIPTION
**Pull request by yegonkimutai**
In this PR, I did the following ;
- [ ] Create models and set-up associations between them
- [ ] Create custom methods to update the counters as well as fetch comments and posts

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,14 @@
 class Comment < ApplicationRecord
-  belongs_to :user_id
-  belongs_to :post_id
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (muneebulrehman) [inline app/models/comment.rb]: - Optional
You don't need to mention `foreign_key` here as you are setting the same in the `User` model. Consider removing it.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+  belongs_to :post
+
+  has_many :likes, dependent: :destroy
```
Comment (muneebulrehman) [inline app/models/comment.rb]: `Comment` has no relation with `Like` as per the ER diagram provided to you. Per the ER diagram, a user will only be able to like a post and not a comment. Kindly remove this part.

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
 class Like < ApplicationRecord
-  belongs_to :user_id
-  belongs_to :post_id
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (muneebulrehman) [inline app/models/like.rb:2]: - Optional
You don't need to mention `foreign_key` here as you are setting the same in the `User` model. Consider removing it.

File: app/models/post.rb [hunk 4]
```diff
@@ -1,3 +1,16 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (muneebulrehman) [inline app/models/post.rb]: - Optional
You don't need to mention `foreign_key` here as you are setting the same in the `User` model. Consider removing it.

File: app/models/post.rb [hunk 5]
```diff
@@ -1,3 +1,16 @@
 class Post < ApplicationRecord
```
Comment (muneebulrehman) [inline app/models/post.rb:1]: There is a relation between a post and likes as per the ER diagram. Kindly add the association between the two as you have done for the `comments`.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @yegonkimutai , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Great job using Rails active callbacks 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 6]
```diff
@@ -1,6 +1,7 @@
...
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :author, class_name: 'User'
   has_many :comments, dependent: :destroy
+  has_many :likes, class_name: 'Like', dependent: :destroy
```
Comment (MahmoudBakr23) [inline app/models/post.rb:4]: OPTIONAL
- Please remove the class_name attribute from the likes association as it's not changing the name so it's already known with the `Like` model

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @yegonkimutai 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done working on the requested changes from the previous reviewer

**Optional**
_See the comment under the review._

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**